Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/orient_polygon_soup.h

```diff
@@ -460,29 +468,57 @@ struct Polygon_soup_orienter
  * @tparam PolygonRange a model of the concept `RandomAccessContainer`
  * whose value_type is a model of the concept `RandomAccessContainer`
  * whose value_type is `std::size_t`.
+ * @tparam NamedParameters a sequence of \ref pmp_namedparameters "Named Parameters"
  *
  * @param points points of the soup of polygons. Some additional points might be pushed back to resolve
  *               non-manifoldness or non-orientability issues.
  * @param polygons each element in the vector describes a polygon using the index of the points in `points`.
  *                 If needed the order of the indices of a polygon might be reversed.
+ * @param np optional sequence of \ref pmp_namedparameters "Named Parameters" among the ones listed below
+ *
+  * \cgalNamedParamsBegin
+  *   \cgalParamBegin{visitor} a class model of `PMPOrientationVisitor`
+  *                            that is used to track the creation of new points.
+  *   \cgalParamEnd
+  * \cgalNamedParamsEnd
+ *
  * @return `true`  if the orientation operation succeded.
  * @return `false` if some points were duplicated, thus producing a self-intersecting polyhedron.
  *
  */
-template <class PointRange, class PolygonRange>
+template <class PointRange, class PolygonRange, class NamedParameters>
 bool orient_polygon_soup(PointRange& points,
-                         PolygonRange& polygons)
+                         PolygonRange& polygons,
+                         const NamedParameters& np)
 {
-  std::size_t inital_nb_pts = points.size();
-  internal::Polygon_soup_orienter<PointRange, PolygonRange> orienter(points, polygons);
+  using parameters::choose_parameter;
+  using parameters::get_parameter;
+
+  typedef typename internal_np::Lookup_named_param_def<
+                                  internal_np::visitor_t,
+                                  NamedParameters,
+                                  internal::Default_PS_orienter_visitor>::type Visitor;
+  Visitor visitor = choose_parameter<Visitor>(get_parameter(np, internal_np::visitor));
```

Comment:
Here the visitor object seems to be passed by copy from the name parameter.

Reply:
We have `get_parameter_reference` now, so that can be fixed, and maybe it is required at other PMP places. As you have said, visitors seem (looking at PMP) mostly taken by reference, so I will align with this.